arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/popup/PopupView.kt

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright 2021 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.esri.arcgisruntime.toolkit.popup
+
+import android.content.Context
+import android.content.res.ColorStateList
+import android.graphics.Color
+import android.text.InputType
+import android.util.AttributeSet
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+import android.widget.EditText
+import android.widget.FrameLayout
+import android.widget.Spinner
+import android.widget.TextView
+import androidx.core.widget.doAfterTextChanged
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.esri.arcgisruntime.ArcGISRuntimeException
+import com.esri.arcgisruntime.data.CodedValueDomain
+import com.esri.arcgisruntime.data.Field
+import com.esri.arcgisruntime.mapping.popup.Popup
+import com.esri.arcgisruntime.mapping.popup.PopupField
+import com.esri.arcgisruntime.mapping.popup.PopupManager
+import com.esri.arcgisruntime.toolkit.BR
+import com.esri.arcgisruntime.toolkit.R
+import kotlinx.android.synthetic.main.item_popup_row.view.*
+import kotlinx.android.synthetic.main.layout_popupview.view.*
+
+private const val TAG = "PopupView"
+
+/**
+ * Displays the popup attribute list in a [RecyclerView].
```

Comment:
PopupViewTestActivity is in the toolkit-test-app module and since it is not a dependency for the arcgis-android-toolkit module, we cannot successfully add the link to PopupViewTestActivity 

>Generally you could add some more information about the implementation of this view component, for example more details about the various inner classes and how they interact with each other. 

Updated.